documentation/modules/cruise-control/con-cruise-control-overview.adoc

```diff
@@ -2,54 +2,508 @@
+
+* *Default goals* refer to the goals used by default when generating proposals. 
+They match the supported goals unless specifically set by the user.
+* *Proposal-specific goals* are a subset of default goals configured for specific proposals.
+
+Configure optimization goals in the `Kafka` and `KafkaRebalance` custom resources.
+
+* `Kafka` resource for supported, hard, and default goals.
+** Supported goals: `Kafka.spec.cruiseControl.config.goals`
+** Hard goals: `Kafka.spec.cruiseControl.config.hard.goals`
+** Default goals: `Kafka.spec.cruiseControl.config.default.goals`
+* `KafkaRebalance` resource for proposal-specific goals.
+** Proposal-specific goals: `KafkaRebalance.spec.goals`
+
+=== Supported goals
+
+Supported goals are predefined and available to use for generating Cruise Control optimization proposals. 
+Goals not listed as supported goals cannot be used in Cruise Control operations.
+Some supported goals are preset as hard goals.
+
+To simplify configuration, use the inherited supported goals unless you need to exclude specific goals from `KafkaRebalance` resources. 
+You can adjust the priority order in the default optimization goals configuration.
+
+Configure supported goals in `Kafka.spec.cruiseControl.config.goals`:
+
+* To accept inherited supported goals, omit the `goals` property.
+* To modify supported goals, specify the goals in descending priority order in the `goals` property.
+
+=== Hard and soft goals
+
+Hard goals must be satisfied for optimization proposals to be generated. 
+Soft goals are best-effort objectives that Cruise Control tries to meet after all hard goals are satisfied.
+The classification of hard and soft goals is fixed in Cruise Control code and cannot be changed.
+
+Cruise Control first prioritizes satisfying hard goals, and then addresses soft goals in the order they are listed.
+A proposal meeting all hard goals is valid, even if it violates some soft goals. 
+
+For example, a soft goal might be to evenly distribute a topic's replicas. 
+Cruise Control continues to generate an optimization proposal even if the soft goal isn't completely satisfied.
+
+Configure hard goals in your Cruise Control deployment using `Kafka.spec.cruiseControl.config.hard.goals`:
+
+* To enforce all hard goals, omit the `hard.goals` property.
+* To specify hard goals, list them in `hard.goals`.
+* To exclude a hard goal, ensure it's not in either `default.goals` or `hard.goals`.
+
+Increasing the number of configured hard goals will reduce the likelihood of Cruise Control generating optimization proposals.
+
+=== Default goals
+
+Cruise Control uses default goals to generate an optimization proposal. 
+You can override default goals by setting proposal-specific optimization goals in a `KafkaRebalance` resource.
+
+If `default.goals` is not specified in the Cruise Control deployment configuration, Strimzi configures `default.goals` to the list of goals specified in `config.goals`. 
```

Comment:
I believe the CC configuration is just `goals` although this field is in the `cruiseControl.config` section of the `Kafka` resource, just like `default.goals` is

```suggestion
If `default.goals` is not specified in the Cruise Control deployment configuration, Strimzi configures `default.goals` to the list of supported goals specified in `goals`. 
```